templates/hotgrid.jsx

```diff
@@ -0,0 +1,108 @@
+import Adapt from 'core/js/adapt';
+import React from 'react';
+import { templates, classes } from 'core/js/reactHelpers';
+
+export default function Hotgrid(props) {
+  const hotgridLabels = Adapt.course.get('_globals')._components._hotgrid;
+  const ariaLabels = Adapt.course.get('_globals')._accessibility._ariaLabels;
+
+  const {
+    _items,
+    _columnWidth,
+    _isRound,
+    onItemClicked
+  } = props;
+
+  const itemAriaLabel = (_index, _graphic, _isVisited) => {
+    const arr = [];
+    const separator = '. ';
+
+    // Show either graphic title or generic title
+    if (_graphic.title) {
+      arr.push(_graphic.title);
+    } else {
+      arr.push(`${hotgridLabels.item} ${_index}`);
+    }
+
+    // Graphic alt text
+    if (_graphic.alt) arr.push(_graphic.alt);
+
+    // Visited state
+    if (_isVisited) arr.push(ariaLabels.visited);
+
+    return { __html: arr.join(separator) };
+  };
+
+  return (
+    <div className='component__inner hotgrid__inner'>
+
+      <templates.header {...props} />
+
+      <div className="component__widget hotgrid__widget">
+
+        <div className="hotgrid__grid" role="list">
+
+          {_items.map(({ _index, _graphic, _isVisited, _isActive }) =>
+
+            <div className="hotgrid__item" role="listitem" key={_index} style={{ width: _columnWidth + '%' }}>
+
+              <button className={classes([
+                'hotgrid__item-btn',
+                `item-${_index}`,
+                (_graphic.srcHover && _graphic.srcVisited) ? 'is-image' : 'is-css',
+                _isRound && 'is-round',
+                _isVisited && 'is-visited',
+                _isActive && 'is-active'
+              ])}
+              onClick={onItemClicked}
+              data-index={_index}
+              >
+
+                <span className="aria-label" dangerouslySetInnerHTML={ itemAriaLabel(_index, _graphic, _isVisited) } />
```

Comment:
Is visited states need to go at the beginning of the label. We're applying this globally but might as well update as we go

Reply:
Moved 'visited' to the beginning.